### PR TITLE
[IMM32] Rewrite ImmSetCompositionWindow

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Implementing Japanese input...
JIRA issue: [CORE-11700](https://jira.reactos.org/browse/CORE-11700)

## Proposed changes

- Rewrite `imm32!ImmSetCompositionWindow` function.

### CODE REVIEW

File: dll/win32/imm32/imm.c [hunk 1]
```diff
@@ -2933,38 +2933,28 @@ BOOL WINAPI ImmSetCompositionStringW(
...
 {
-    BOOL reshow = FALSE;
-    InputContextData *data = get_imc_data(hIMC);
-
-    TRACE("(%p, %p)\n", hIMC, lpCompForm);
-    if (lpCompForm)
-        TRACE("\t%x, %s, %s\n", lpCompForm->dwStyle,
-              wine_dbgstr_point(&lpCompForm->ptCurrentPos),
-              wine_dbgstr_rect(&lpCompForm->rcArea));
+    DWORD idImeThread, idThread;
```
Comment (HBelusca) [inline dll/win32/imm32/imm.c:2936]: similar to other PR: `dwThreadId`, `dwImeThreadId` ..

File: dll/win32/imm32/imm.c [hunk 2]
```diff
@@ -2933,38 +2933,28 @@ BOOL WINAPI ImmSetCompositionStringW(
...
+    LPINPUTCONTEXT pIC;
+    HWND hWnd;
 
-    if (!data)
-    {
-        SetLastError(ERROR_INVALID_HANDLE);
+    idImeThread = NtUserQueryInputContext(hIMC, 1);
```
Comment (HBelusca) [inline dll/win32/imm32/imm.c:2940]: cf. remark about this magic "1" from other PR review.

File: dll/win32/imm32/imm.c (hunk 1 above)
Comment (katahiromz) [reply dll/win32/imm32/imm.c]: Renamed. OK.

File: dll/win32/imm32/imm.c (hunk 2 above)
Comment (katahiromz) [reply dll/win32/imm32/imm.c]: Hypothesis: `Imm32QueryInputContext`'s 2nd paramter is same as `Imm32QueryWindow`'s 2nd parameter.

**Pull request merged**